ocaml/typing/typetexp.ml

```diff
@@ -525,10 +525,10 @@ and transl_type_aux env policy mode styp =
       | Error e ->
         raise (Error(ctyp_loc, env,
                      Non_value {vloc = Tuple; err = e; typ = ctyp_type})))
-      ctys;
+      labeled_ctys;
     (* CR labeled tuples: handle labeled tuple type expressions *)
-    let ty = newty (Ttuple (List.map (fun ctyp -> None, ctyp.ctyp_type) ctys)) in
-    ctyp (Ttyp_tuple ctys) ty
+    let ty = newty (Ttuple (List.map (fun (label, ctyp) -> label, ctyp.ctyp_type) labeled_ctys)) in
+    ctyp (Ttyp_tuple (List.map snd labeled_ctys)) ty
```

Comment:
I think this PR should either update `core_type_desc` to have labels or error here - it feels wrong for the `core_type_desc` and the `type_desc` to disagree in this way (though I'm struggling to come up with an actual example of bad behavior).

I suspect it's not much work to add the labels; `core_type_desc` isn't used for much.